File: backend/schemas/calendar_note.py
from datetime import datetime, date
from pydantic import BaseModel


class CalendarNoteCreate(BaseModel):
    title: str
    content: str | None = None
    note_date: date
    color: str | None = "#6D4CFF"
    is_pinned: bool = False
    is_all_day: bool = True


class CalendarNoteUpdate(BaseModel):
    title: str | None = None
    content: str | None = None
    note_date: date | None = None
    color: str | None = None
    is_pinned: bool | None = None
    is_all_day: bool | None = None


class CalendarNoteResponse(BaseModel):
    id: int
    title: str
    content: str | None = None
    note_date: date
    color: str | None = "#6D4CFF"
    is_pinned: bool = False
    is_all_day: bool = True
    user_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: web/src/types/CalendarNote.ts
export interface CalendarNote {
  id: number;
  title: string;
  content?: string | null;
  note_date: string; // YYYY-MM-DD
  color?: string | null;
  is_pinned?: boolean;
  is_all_day?: boolean;
  user_id?: number;
  created_at?: string;
  updated_at?: string;
}

export interface CalendarNoteCreate {
  title: string;
  content?: string | null;
  note_date: string;
  color?: string | null;
  is_pinned?: boolean;
  is_all_day?: boolean;
}

export interface CalendarNoteUpdate {
  title?: string;
  content?: string | null;
  note_date?: string;
  color?: string | null;
  is_pinned?: boolean;
  is_all_day?: boolean;
}
